### PR TITLE
perf: Use db.session.query instead of model query 

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to test for performance improvements when using flask-sqlalchemy's db.session.query instead of model.query.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by kruai** [comment]
FYI, I'm going to self-merge this to test the performance difference in Stage, and will revert if it gets worse.

File: api/host_query_db.py [hunk 1]
```diff
@@ -172,7 +172,8 @@ def _order_how(column, order_how: str):
 def _find_all_hosts(columns: List[ColumnElement] = None) -> Query:
     identity = get_current_identity()
     query = (
-        Host.query.join(HostGroupAssoc, isouter=True)
+        db.session.query(Host)
```
Comment (vkrizan) [inline api/host_query_db.py:175]: Do you have more info on what would this bring?

Could this lead to memory leak? Or, what is the lifespan of the session.

Comment (kruai) [reply api/host_query_db.py:175]: Apparently the performance is generally better when querying with db.session. Apparently flask-sqlalchemy has automatic handling for these, but I also added `db.session.close()` after each place I actually fetch the data. For example: https://github.com/RedHatInsights/insights-host-inventory/blob/master/api/host_query_db.py#L56

Comment (kruai) [reply api/host_query_db.py:175]: Honestly, the performance change wasn't noticeable, so I'm fine with either implementation. I just wanted to figure out any potential ways to improve performance

**Pull request merged**